src/axom/quest/examples/delaunay_triangulation.cpp

```diff
@@ -0,0 +1,231 @@
+    if(boundsMin.empty())
+    {
+      boundsMin.clear();
+      boundsMax.clear();
+      for(int i = 0; i < dimension; ++i)
+      {
+        boundsMin.push_back(0.);
+        boundsMax.push_back(1.);
+      }
+    }
+
+    SLIC_INFO(axom::fmt::format(R"(Using parameter values:
+    {{
+      dimension: {}
+      nrandpt: {}
+      bounding box min: {{{}}}
+      bounding box max: {{{}}}
+      outfile = '{}'
+      intermediate output steps: {}
+    }})",
+                                dimension,
+                                numRandPoints,
+                                axom::fmt::join(boundsMin, ", "),
+                                axom::fmt::join(boundsMax, ", "),
+                                outputVTKFile,
+                                numOutputSteps));
+  }
+};
+
+/// Utility function to create Delaunay Triangulation over random points
+template <int DIM>
+void run_delaunay(const Input& params)
+{
+  using axom::utilities::random_real;
+  using axom::utilities::string::removeSuffix;
+
+  using Delaunay = axom::quest::Delaunay<DIM>;
+  using PointType = typename Delaunay::PointType;
+  using BoundingBox = typename Delaunay::BoundingBox;
+
+  const int numPoints = params.numRandPoints;
+  const int numOutputVTKsteps = params.numOutputSteps;
+  const std::string& outputVTKFile = removeSuffix(params.outputVTKFile, ".vtk");
+
+  // Use a slam::ModularInt to help with bookkeeping
+  const int dumpFreq =
+    params.shouldOutputSteps() ? numPoints / numOutputVTKsteps : numPoints;
+  axom::slam::ModularInt<> dumperMod(0, dumpFreq);
+
+  BoundingBox bbox {PointType(params.boundsMin.data()),
+                    PointType(params.boundsMax.data())};
+
+  axom::utilities::Timer timer(true);
+
+  // Create initial Delaunay triangulation over bounding box
+  Delaunay dt;
+  dt.initializeBoundary(bbox);
+
+  // Incrementally insert random points within bounding box
+  for(int i = 0; i < numPoints; ++i)
+  {
+    PointType new_pt;
+    for(int d = 0; d < DIM; ++d)
+    {
+      new_pt[d] = random_real(bbox.getMin()[d], bbox.getMax()[d]);
+    }
+
+    // Insert the point into the triangulation
+    dt.insertPoint(new_pt);
+
+    // Optionally, dump an intermediate mesh file
+    if(params.shouldOutputSteps() && (dumperMod++ == 0 || i == numPoints - 1))
+    {
+      std::string fname = axom::fmt::format("{}_{:06}.vtk", outputVTKFile, i);
+      dt.writeToVTKFile(fname);
+    }
+  }
+
+  //Remove the starting rectangular box
+  dt.removeBoundary();
+
+  timer.stop();
+
+  SLIC_INFO(axom::fmt::format(
+    "It took {} seconds to create a Delaunay complex with {} "
+    "points. Mesh has {} {}. Insertion rate of {:.1f} points per second.",
+    timer.elapsedTimeInSec(),
+    dt.getMeshData()->getNumberOfValidVertices(),
+    dt.getMeshData()->getNumberOfValidElements(),
+    DIM == 2 ? "triangles" : "tetrahedra",
+    numPoints / timer.elapsedTimeInSec()));
+  axom::slic::flushStreams();
```

Comment:
I'm reasonably certain flushing on a `SimpleLogger` doesn't do anything because it immediately writes to `std::cout`.  Did you see a behavior that needed this or was it just added because most loggers do need to be flushed?

Reply:
I think it's the latter, and I got in the habit of flushing before expensive operations.
I removed these.